File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.py
def solution(s):
    answer = 0
    
    def find1(lt,rt):
        while lt>=0 and rt<len(s) and s[lt] == s[rt]:
            lt-=1
            rt+=1
        return rt-lt-1
    
    def find2(lt,rt):
        while lt>=0 and rt<len(s) and s[lt] == s[rt]:
            lt-=1
            rt+=1
        return rt-lt-1
    
    
    for i in range(len(s)):
        len1 = find1(i,i)
        len2 = find2(i,i+1)

        c = max(len1, len2)
        answer = max(answer, c)
    
    return answer

File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.java
import java.util.*;
class Solution
{   static String s;
    public int solution(String s)
    {
        
        int answer = 1;
        
        for(int i=0;i<s.length();i++){
            answer = Math.max(answer, find(s, i, i));
            answer = Math.max(answer, find(s,i,i+1));
            
        }

        return answer;
    }
 public static int find(String s, int lt, int rt){
     while(lt>=0 && rt<s.length()&& s.charAt(lt)== s.charAt(rt)){
            lt--;
            rt++;
        }
     return rt-lt-1;
 }
 }